wavetorch/core/cell.py

```diff
@@ -203,14 +207,22 @@ def step(self, x, y1, y2, c_linear, proj_rho):
         else:
             c = c_linear
 
-        y = torch.mul((dt**(-2) + b * 0.5 * dt**(-1)).pow(-1),
+
+        y = torch.mul(
+            (dt**(-2) + b * 0.5 * dt**(-1)).pow(-1),
                       (2/dt**2*y1 - torch.mul( (dt**(-2) - b * 0.5 * dt**(-1)), y2)
                                + torch.mul(c.pow(2), conv2d(y1.unsqueeze(1), self.laplacian, padding=1).squeeze(1)))
                      )
         
         # Insert the source
-        y[:, self.src_x, self.src_y] = y[:, self.src_x, self.src_y] + x.expand_as(y[:, self.src_x, self.src_y])
+        # Orginal code (v0.1)
+        # y[:, self.src_x, self.src_y] = y[:, self.src_x, self.src_y] + x.expand_as(y[:, self.src_x, self.src_y])
         
+        # Modified for pytorch > 1 (inplace bug in orginal)
+        x_in = torch.zeros(y.size()).detach()
+        x_in[:, self.src_x, self.src_y] = x
+        
+        y = y + x_in
```

Comment:
Can you clarify here what the issue was with the original code? Is this something that should work on pytorch 1.0 but not on 1.2? I don't seem to be getting any explicit errors on either. Or is the issue that the gradients are just wrong?

Reply:
Nevermind, I was able to reproduce the issue.

Reply:
This appears to be an issue only when using a line source. I'm guessing you encountered it when running `propagate.py` but probably not when running the vowel training?

Reply:
Exactly.